OsmAndMapCreator/src/net/osmand/BinaryComparator.java

```diff
@@ -41,40 +48,45 @@
 	private static final int STREET_NAME_COMPARE = 22;
 	private static final int BUILDINGS_COMPARE = 31;
 	private static final int INTERSECTIONS_COMPARE = 41;
+	private static final int POI_COMPARE = 51;
+	private static final int COMPARE_ADD = 91;
+	private static final int COMPARE_RM = 92;
 	private static final int OSM_OUTPUT = -1;
+	private static final int[] ADDRESS_COMPARE =
+			{CITY_COMPARE, CITY_NAME_COMPARE, STREET_COMPARE, STREET_NAME_COMPARE, BUILDINGS_COMPARE, INTERSECTIONS_COMPARE};
 	private static final Map<String, Integer> COMPARE_ARGS = new HashMap<String, Integer>() {{
 		put("--cities", CITY_COMPARE);
 		put("--city-names", CITY_NAME_COMPARE);
 		put("--streets", STREET_COMPARE);
 		put("--street-names", STREET_NAME_COMPARE);
 		put("--buildings", BUILDINGS_COMPARE);
 		put("--intersections", INTERSECTIONS_COMPARE);
+		put("--poi", POI_COMPARE);
+		put("--add", COMPARE_ADD);
```

Comment:
The parameters rm, add have very confusing names. Better to use, --ignore-missing-1 and --ignore-missing-2 of course it should be applicable for address search.